src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx

```diff
@@ -242,117 +210,54 @@ export const ReviewSpendingLimits = ({ onBack, onClose, txToken, values }: Revie
     getResetTimeOptions().find(({ value }) => value === (+existentSpendingLimit.resetTimeMin).toString())?.label ??
     'One-time spending limit'
 
-  const closeEditModalCallback = (txParameters: TxParameters) => {
-    const oldGasPrice = gasPriceFormatted
-    const newGasPrice = txParameters.ethGasPrice
-    const oldSafeTxGas = gasEstimation
-    const newSafeTxGas = txParameters.safeTxGas
-    const newSafeNonce = txParameters.safeNonce
-
-    if (newGasPrice && oldGasPrice !== newGasPrice) {
-      setManualGasPrice(txParameters.ethGasPrice)
-    }
-
-    if (txParameters.ethGasLimit && gasLimit !== txParameters.ethGasLimit) {
-      setManualGasLimit(txParameters.ethGasLimit)
-    }
-
-    if (newSafeTxGas && oldSafeTxGas !== newSafeTxGas) {
-      setManualSafeTxGas(newSafeTxGas)
-    }
-
-    if (newSafeNonce) {
-      const newSafeNonceNumber = parseInt(newSafeNonce, 10)
-      setManualSafeNonce(newSafeNonceNumber)
-    }
-  }
-
-  let confirmButtonText = 'Submit'
-  if (ButtonStatus.LOADING === buttonStatus) {
-    confirmButtonText = txEstimationExecutionStatus === EstimationStatus.LOADING ? 'Estimating' : 'Submitting'
-  }
-
   return (
-    <EditableTxParameters
-      isOffChainSignature={isOffChainSignature}
-      isExecution={canTxExecute}
-      ethGasLimit={gasLimit}
-      ethGasPrice={gasPriceFormatted}
-      safeTxGas={gasEstimation}
-      closeEditModalCallback={closeEditModalCallback}
+    <TxParamsState
+      txData={estimateGasArgs.txData || ''}
```

Comment:
Is this ok to fallback to a string?

Reply:
I assume so as it is defaulted to '' in other modals. See https://github.com/gnosis/safe-react/blob/42c2ee62a4368a869cf68864b06d8353b6b08336/src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/Review/index.tsx#L68-L72